src/client/auth.ts

```diff
@@ -478,6 +497,178 @@ export function extractResourceMetadataUrl(res: Response): URL | undefined {
   }
 }
 
+/**
+ * Extracts required scopes from WWW-Authenticate header following SEP-835.
+ * 
+ * This function parses the WWW-Authenticate header to extract OAuth scopes
+ * that should be requested when authenticating. According to SEP-835, 
+ * this provides contextual guidance on the minimum scopes needed.
+ * 
+ * @param res - HTTP response containing WWW-Authenticate header
+ * @returns Array of required scope strings, or undefined if not present
+ */
+export function extractScopesFromWwwAuthenticate(res: Response): string[] | undefined {
+  const authenticateHeader = res.headers.get("WWW-Authenticate");
+  if (!authenticateHeader) {
+    return undefined;
+  }
+
+  const [type, scheme] = authenticateHeader.split(' ');
+  if (type.toLowerCase() !== 'bearer' || !scheme) {
+    return undefined;
+  }
```

Comment:
This is duplicated code from existing `extractResourceMetadataUrl` in this same file. Maybe worth refactoring in a dedicated `parseWwwAuthenticateHeader` ?